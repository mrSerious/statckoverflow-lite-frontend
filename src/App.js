import React, { Fragment, Component } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
import { Provider } from 'react-redux';
import { persistor, store } from './store/store';
import NotFound from './components/404/NotFound';
import routes from './routes';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Provider store={store}>
        <PersistGate persistor={persistor}>
          <Router>
            <Fragment>
              <Switch>
                {
                  routes.map(route => (
                    <Route
                      exact={route.exact}
                      path={route.path}
                      key={route.path}
                      component={route.component}
                      />
                  ))
                }
                <Route component={NotFound} />
              </Switch>
            </Fragment>
          </Router>
        </PersistGate>
      </Provider>
    );
  }
}

export default App;
