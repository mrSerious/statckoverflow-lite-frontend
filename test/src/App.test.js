import React from 'react';
import { shallow } from 'enzyme';

import App from '../../src/App';

test('App snapshot test', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
