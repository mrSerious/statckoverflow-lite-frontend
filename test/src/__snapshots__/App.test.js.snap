// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <Connect(withRouter(Header)) />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          key="/"
          path="/"
        />
        <Route
          component={[Function]}
          exact={true}
          key="/results"
          path="/results"
        />
        <Route
          component={[Function]}
          exact={true}
          key="/signup"
          path="/signup"
        />
        <Route
          component={[Function]}
          exact={true}
          key="/login"
          path="/login"
        />
        <Route
          component={[Function]}
          exact={true}
          key="/post-question"
          path="/post-question"
        />
        <Route
          component={[Function]}
        />
      </Switch>
      <Footer />
      <Connect(ReduxToastr)
        closeOnToastrClick={true}
        newestOnTop={true}
        position="top-right"
        preventDuplicates={true}
        progressBar={true}
        timeOut={6000}
        transitionIn="fadeIn"
        transitionOut="fadeOut"
      />
    </BrowserRouter>
  </PersistGate>
</Provider>
`;
